recipes-automotive/sdl-generic-hmi/sdl-generic-hmi_git.bb

```diff
@@ -0,0 +1,33 @@
+SUMMARY = "HTML based reference HMI for SmartDeviceLink"
+DESCRIPTION = "SmartDeviceLink (SDL) is a standard set of protocols and \
+messages that connect applications on a smartphone to a vehicle head \
+unit."
+HOMEPAGE = "https://www.smartdevicelink.com"
+SECTION = "app"
+
+LICENSE = "BSD-3-Clause"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=cf455e8d39d2ded1d85b1c5dea6ec3c5"
+
+SRC_URI = "git://github.com/smartdevicelink/generic_hmi.git;branch=develop"
+
+PV = "0.0.0+git${SRCPV}"
+SRCREV = "96c44b8f488977fe1cf29af1a20898187a982b63"
+
+S = "${WORKDIR}/git"
+
+PACKAGES = " \
+    ${PN} \
+"
+
+FILES_${PN}_append = " \
+    ${datadir}/smartdevicelink/generic_hmi/ \
+"
+
+do_install() {
+    install -d ${D}${datadir}/smartdevicelink/generic_hmi
+    cp -a ${S}/index.html ${D}${datadir}/smartdevicelink/generic_hmi/
+    cp -a ${S}/build ${D}${datadir}/smartdevicelink/generic_hmi/
```

Comment:
Please use 'install' instead of 'cp'. There is warning here

sdl-generic-hmi-0.0.0+gitAUTOINC+96c44b8f48: sdl-generic-hmi: /sdl-generic-hmi/usr/share/smartdevicelink/generic_hmi/index.html is owned by uid 1000, which is the same as the user running bitbake. This may be due to host contamination
sdl-generic-hmi: /sdl-generic-hmi/usr/share/smartdevicelink/generic_hmi/build/bundle.js is owned by uid 1000, which is the same as the user running bitbake. This may be due to host contamination [host-user-contaminated]

Reply:
Fixed in update, please take a look if this is ok.

Reply:
OK, thanks for checking.  I'm aware that install is preferred, but did not know it caused a user id warning/error but that might be new.  If I recall correctly it might have been because the hierarchy of files/directories was large, so a recursive copy made more sense than to list each and every one.  I wonder, is that the only/preferred method for all bitbake jobs.  It is more explicit of course, but also a bit cumbersome.  (There's no recursive flag on "install" command, correct?)

Reply:
Yes, "install" no recursive flag. The "install"  is recommended. There is a style guide in wiki of openembedded [Styleguide](https://www.openembedded.org/wiki/Styleguide#do_install)